apilib/src/main/java/io/swagger/client/ApiClient.java

```diff
@@ -103,20 +112,24 @@ public ApiClient(String authName, String clientId, String secret, String usernam
                 .setPassword(password);
     }
 
-   public void createDefaultAdapter() {
+    public void createDefaultAdapter() {
         Gson gson = new GsonBuilder()
                 .setDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSZ")
+                .registerTypeAdapter(DateTime.class, new DateTimeTypeAdapter())
+                .registerTypeAdapter(LocalDate.class, new LocalDateTypeAdapter())
                 .create();
 
-       OkHttpClient.Builder builder = new OkHttpClient.Builder();
-       for (Map.Entry<String, Interceptor> entry : apiAuthorizations.entrySet())
-       {
-           builder.addInterceptor(entry.getValue());
-       }
-       HttpLoggingInterceptor httpLoggingInterceptor = new HttpLoggingInterceptor();
-       httpLoggingInterceptor.setLevel(HttpLoggingInterceptor.Level.BODY);
-       builder.addInterceptor(httpLoggingInterceptor); // LOGS request and response
-       okClient = builder.build();
+        okBuilder = new OkHttpClient.Builder();
+
+        // [DEV NOTE: Added custom code to add interceptors for authorizations]
+        for (Map.Entry<String, Interceptor> entry : apiAuthorizations.entrySet()) {
```

Comment:
TODO: I still need to do this. So, might need to open issue or create a PR to fix the template.